tests/tests/test_compat.py

```diff
@@ -189,40 +189,68 @@ def assert_successful_deployment(api_deployments, deployment_id, timeout=TIMEOUT
             time.sleep(1)
 
 
-def _watch_clients(host_addr, clients, timeout=TIMEOUT):
+class RebootDetectorThread(threading.Thread):
+    def __init__(self, host_ip, client, timeout, barrier, **kwargs):
+        super().__init__(**kwargs)
+        self._host_ip = host_ip
+        self._client = client
+        self._timeout = timeout
+        self._exception_q = queue.Queue(maxsize=1)
+        self.detector = self._client.get_reboot_detector(self._host_ip)
+        self.barrier = barrier
+
+    def run(self):
+        try:
+            # Setup the individual reboot-detectors
+            with self.detector as rd:
+                # Possibly move this to after the reboot-detector accepts connections (?)
```

Comment:
... I say yes, and then time out if the client does not respond (ref. review comment).

Reply:
I think I'm going to leave it as it is actually. This is easier, and cleaner. It fixes the recurring issue we have been seeing. The comment is a little misleading now, as it was made before some of the changes.